lib/Sema/CSDiagnostics.cpp

```diff
@@ -7002,3 +7002,21 @@ bool MissingContextualTypeForNil::diagnoseAsError() {
   emitDiagnostic(diag::unresolved_nil_literal);
   return true;
 }
+
+bool ReferenceToInvalidDeclaration::diagnoseAsError() {
+  auto *decl = castToExpr<DeclRefExpr>(getAnchor())->getDecl();
+  assert(decl);
+
+  auto &DE = getASTContext().Diags;
+  // This problem should have been already diagnosed during
+  // validation of the declaration.
+  if (DE.hadAnyError())
+    return true;
+
+  // If no errors have been emitted yet, let's emit one
+  // about reference to an invalid declaration.
+
+  emitDiagnostic(diag::reference_to_invalid_decl, decl->getName());
```

Comment:
When would the solver actually emit this error?

Reply:
In cases when actual problem with declaration hasn't been diagnosed due to a bug (hopefully never).

Reply:
I was thinking about adding bit about asking to report it just like we do with actual fallback diagnostic.